### PR TITLE
Metabase lib cleanup - metadata edition

### PR DESCRIPTION
**Pull request by tlrobinson**

### CODE REVIEW

File: frontend/src/metabase-lib/lib/metadata/Database.js [hunk 1]
```diff
@@ -33,6 +35,8 @@ export default class Database extends Base {
     return this.name;
   }
 
+  // SCEMAS
```
Comment (paulrosenzweig) [inline frontend/src/metabase-lib/lib/metadata/Database.js:38]:
```suggestion
  // SCHEMAS
```

File: frontend/src/metabase-lib/lib/metadata/Database.js [hunk 2]
```diff
@@ -41,6 +45,20 @@ export default class Database extends Base {
...
+
+  @memoize
+  tablesLookup() {
+    return createLookupByProperty(this.tables, "id");
+  }
+
+  // @deprecated: use tablesLookup
+  get tables_lookup() {
+    return this.tablesLookup();
+  }
```
Comment (paulrosenzweig) [inline frontend/src/metabase-lib/lib/metadata/Database.js:58]: I like the approach of swapping out the properties for getters labeled as deprecated. Could we remove them entirely here?

File: frontend/src/metabase/lib/query/field_ref.js [hunk 3]
```diff
@@ -139,11 +138,11 @@ export function getFieldTarget(field, tableDef, path = []) {
...
-    fieldDef.filter_operators = getFilterOperators(fieldDef, tableDef);
-    fieldDef.filter_operators_lookup = createLookupByProperty(
-      fieldDef.filter_operators,
-      "name",
-    );
+    // fieldDef.filter_operators = getFilterOperators(fieldDef, tableDef);
+    // fieldDef.filter_operators_lookup = createLookupByProperty(
+    //   fieldDef.filter_operators,
+    //   "name",
+    // );
```
Comment (paulrosenzweig) [inline frontend/src/metabase/lib/query/field_ref.js:145]: can be removed

File: frontend/test/metabase/lib/expressions/compile.unit.spec.js [hunk 4]
```diff
@@ -6,7 +6,7 @@ import {
   expressionOpts,
 } from "./__support__/expressions";
 
-const ENABLE_PERF_TESTS = !process.env["CI"];
+const ENABLE_PERF_TESTS = false; //!process.env["CI"];
```
Comment (paulrosenzweig) [inline frontend/test/metabase/lib/expressions/compile.unit.spec.js:9]: Were you just stopping these from running locally temporarily or do you want to remove them?

File: frontend/src/metabase/query_builder/components/dataref/TablePane.jsx [hunk 5]
```diff
@@ -1,85 +1,85 @@
...
-      .catch(error => {
-        this.setState({
-          error: t`An error occurred loading the table`,
-        });
+  async componentWillMount() {
+    try {
+      await Promise.all([
+        this.props.fetchForeignKeys({ id: this.props.tableId }),
+        this.props.fetchMetadata({ id: this.props.tableId }),
+      ]);
```
Comment (paulrosenzweig) [inline frontend/src/metabase/query_builder/components/dataref/TablePane.jsx:55]: Not worth doing here, but what do you think about having object loaders take `objectActionLoaders: ['fetchMetadata']` or something that waits on object actions before removing the loading spinner?

**Review by paulrosenzweig — APPROVED** [review]

File: frontend/src/metabase/query_builder/components/dataref/TablePane.jsx (hunk 5 above)
Comment (tlrobinson) [reply frontend/src/metabase/query_builder/components/dataref/TablePane.jsx:55]: Something like that might make sense, but currently we use the `getLoading` selector which I think is more robust than waiting on the promise that's returned. I'll see what I can come up with.

**Pull request merged**